### ISSUE TITLE
ec2_securitygroup: Could not set 'present' on ensure: You are not authorized to perform this operation.

### ISSUE THREAD

**Issue by jsnod** [puppetlabs/puppetlabs-aws#278]
I am using an IAM account with the policy from https://github.com/puppetlabs/puppetlabs-aws/tree/master/examples/iam-profile/ but I am unable to create a new Security Group:

```
[root@puppet ~] puppet agent -t
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Info: Caching catalog for puppet.domain.com
Info: Applying configuration version '1455064983'
Info: Checking if security group PuppetlabsAWS-test exists in region us-west-2
Info: Creating security group PuppetlabsAWS-test in region us-west-2
Error: Could not set 'present' on ensure: You are not authorized to perform this operation. at 25:/etc/puppet/environments/production/modules/aws_domain/manifests/securitygroups.pp
Error: Could not set 'present' on ensure: You are not authorized to perform this operation. at 25:/etc/puppet/environments/production/modules/aws_domain/manifests/securitygroups.pp
Wrapped exception:
You are not authorized to perform this operation.
Error: /Stage[main]/Aws_domain::Securitygroups/Ec2_securitygroup[PuppetlabsAWS-test]/ensure: change from absent to present failed: Could not set 'present' on ensure: You are not authorized to perform this operation. at 25:/etc/puppet/environments/production/modules/aws_domain/manifests/securitygroups.pp
Notice: Finished catalog run in 25.27 seconds
```

I get the same when I try from the command line using resources:

```
[root@puppet ~] puppet resource ec2_securitygroup test-group ensure=present description="testing" region=us-west-2
Error: Could not set 'present' on ensure: You are not authorized to perform this operation.
Error: Could not set 'present' on ensure: You are not authorized to perform this operation.
Wrapped exception:
You are not authorized to perform this operation.
Error: /Ec2_securitygroup[test-group]/ensure: change from absent to present failed: Could not set 'present' on ensure: You are not authorized to perform this operation.
ec2_securitygroup { 'test-group':
  ensure => 'absent',
}
```

However, when I _list_ Security Groups with `puppet resource ec2_securitygroup`, it works just fine and I see all of my Security Groups.  This leads me to believe that it's not an issue with permissions or my IAM, but something else.  What am I doing wrong?  Thx!

**Comment by kraigh** [comment]
I'm experiencing the same issue. Is there anything needed to inform the module to use the IAM role?

**Comment by jsnod** [comment]
Found another hint in the Puppet logs:

```Could not prefetch ec2_securitygroup provider 'v2': unable to sign request without credentials set```

Which is weird because the credentials are set to _view_ the list of Security Groups just fine.

**Comment by daveseff** [comment]
I ended up putting my keys in $::confdir/puppetlabs_aws_credentials.ini. 

    [default]
      aws_access_key_id = XXXXXXXXXXXXXXXX
      aws_secret_access_key = yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy

It's poorly documented, but you'll need that conf file plus $::confdir/puppetlabs_aws_configuration.ini also with a [default] section.

**Comment by jsnod** [comment]
Still getting the same error, even when using ini files.

I was suspecting that maybe this is related to https://github.com/puppetlabs/puppetlabs-aws/issues/224 but I tried a minimal profile and still getting the same error.

**Comment by szemlyanoy** [comment]
Approach with ini file helped me

**Comment by jsnod** [comment]
I was able to finally solve this.  Turns out the IAM profile.json at https://github.com/puppetlabs/puppetlabs-aws/blob/master/examples/iam-profile/profile.json has a typo.  The action should be a singular `"ec2:CreateSecurityGroup",` rather than the plural `"ec2:CreateSecurityGroups",`

Reference: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateSecurityGroup.html

**Comment by zleswomp** [comment]
@jsnod Would you mind sending a PR to fix the example?  I believe that would close this out.

**Comment by dallinb** [comment]
Before I saw this issue had already been raised, I've raised #337.

**Comment by jsnod** [comment]
Thanks @dallinb! I had been meaning to submit a PR but have been swamped.

**Comment by dallinb** [comment]
No worries :smile:

**Comment by dallinb** [comment]
I think that this issue can now be closed now that PR #337 has been merged.  Does @jsnod agree?

**Comment by ahenroid** [comment]
@jsnod @dallinb, it sounds like this was fixed by #337, so I'm going to close this issue unless I hear otherwise. Thanks!
